scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -0,0 +1,346 @@
+      }
+      tempPositionList.add(containerInfo);
+      dto.setPositionList(tempPositionList);
+    };
+  }
+
+  /**
+   * Extract a string expression from the metadata file and write it into the DTO.
+   *
+   * @param expression the XPath expression yielding the string
+   * @param setter writer for the string
+   * @return
+   */
+  private static RevioPacBioProcessor.ProcessMetadata processString(
+      String expression, BiConsumer<PacBioNotificationDto, String> setter) {
+    XPathExpression expr = RunProcessor.compileXPath(expression)[0];
+    return (document, dto, timeZone) -> {
+      String result = (String) expr.evaluate(document, XPathConstants.STRING);
+      if (result != null) {
+        setter.accept(dto, result);
+      }
+    };
+  }
+
+  public RevioPacBioProcessor(Builder builder, String address) {
+    super(builder);
+  }
+
+  public static RunProcessor create(Builder builder, ObjectNode parameters) {
+    JsonNode address = parameters.get("address");
+    return address.isTextual()
+        ? new RevioPacBioProcessor(builder, address.textValue().replaceAll("/+$", ""))
+        : null;
+  }
+
+  @Override
+  public Stream<File> getRunsFromRoot(File root) {
+    return Arrays.stream(
+        Objects.requireNonNull(
+            root.listFiles(f -> f.isDirectory() && RUN_DIRECTORY.matcher(f.getName()).matches())));
+  }
+
+  @Override
+  public NotificationDto process(File runDirectory, TimeZone tz) throws IOException {
+    // We create one DTO for a run, but there are going to be many wells with
+    // independent and
+    // duplicate metadata that will simply
+    // overwrite in the shared DTO. If the data differs, the last well wins.
+    PacBioNotificationDto dto = new PacBioNotificationDto();
+    dto.setPairedEndRun(false);
+    dto.setSequencerFolderPath(runDirectory.getAbsolutePath());
+
+    // Get the number of SMRT Cells in the run directory
+    int smrtCellCount = (int) getCellDirectory(runDirectory).count();
+    dto.setLaneCount(smrtCellCount);
+
+    // Presence of Transfer_Test_*.txt indicates the run has started, get the creation time
+    Optional<File> mostRecentTransferTestFile =
+        getMetadataDirectory(runDirectory)
+            .flatMap(
+                metadataDirectory ->
+                    Stream.of(
+                        Objects.requireNonNull(
+                            metadataDirectory.listFiles(
+                                file -> TRANSFER_TEST.test(file.getName())))))
+            .min(Comparator.comparing(RevioPacBioProcessor::getFileCreationTime));
+
+    // Set completion time based on most recently created .Transfer_Test file
+    dto.setStartDate(
+        getFileCreationTime(Objects.requireNonNull(mostRecentTransferTestFile.orElse(null))));
+
+    // Set completion time based on most recently created .Transfer_Test file
+    dto.setStartDate(
+        getFileCreationTime(Objects.requireNonNull(mostRecentTransferTestFile.orElse(null))));
```

Comment:
It looks like these two calls are identical, and do a different thing than the comment indicates.